### DISCUSSION TITLE
Does Firecracker really not give memory back to the host?

### DISCUSSION THREAD

**Discussion by ringods** [firecracker-microvm/firecracker#3956] [Q&A]
Within a few weeks, I am going to test running Firecracker on a k3s cluster (homelab) for better isolation of my workloads. In my case, these are general purpose, long running workloads. 

Earlier information collection helped me find #1571 about virtio memory ballooning.

Now, the following article is going around:

https://hocus.dev/blog/qemu-vs-firecracker/

One of the footnotes reads:

> If you squint hard enough, you'll find that Firecracker does support dynamic memory management with a technique called ballooning. However, in practice, it's not usable. To reclaim memory, you need to make sure that the guest OS isn't using it, which, for a general-purpose workload, is nearly impossible.

Can someone shed light on the applicability of using Firecracker to run general purpose workloads?

**Comment by kalyazin** [comment]
@ringods thanks for your question. That is right, Firecracker was designed to work well with ephemeral compute workloads. Handling long running workloads has not been among our main objectives, so presently there is no efficient mechanism in Firecracker for returning memory that is not used by the guest to the host.
